src/clear/model/json_serializable.cr

```diff
@@ -0,0 +1,71 @@
+macro columns_to_instance_vars
+  struct Assigner
+    include JSON::Serializable
+
+    {% for name, settings in COLUMNS %}
+      getter {{name.id}} : {{settings[:type]}}?
+    {% end %}
+
+    def to_create
+      parse_from_json({{@type}}.new) # Loop through instance variables and assign to the newly created orm instance
+    end
+
+    def to_update(to_update_model)
+      parse_from_json(to_update_model) # Loop through instance variables and assign to the orm instance you are updating
+    end
+
+    macro finished
+      def parse_from_json(model)
+        {% for name, settings in COLUMNS %}
+          model.{{name.id}} = @{{name.id}}.not_nil! unless @{{name.id}}.nil?
+        {% end %}
+
+        model
+      end
+    end
+  end
+
+  def self.from_json(string_or_io : String | IO)
+    Assigner.from_json(string_or_io)
+  end
+
+  def self.new(string_or_io : String | IO)
+    Assigner.from_json(string_or_io).to_create
+  end
+
+  def self.create(string_or_io : String | IO)
+    self.new(string_or_io).save
+  end
+
+  def self.create!(string_or_io : String | IO)
+    self.new(string_or_io).save!
+  end
```

Comment:
I'm not sure if these should be included. It's not explicit that it's coming from JSON.

Reply:
See comment below